File: TPs/TP1/Palindromo.java
public class Palindromo {
    public static void main(String[] args) {

        while (true) {
            String frase = MyIO.readLine();
            boolean ispalind = true;
            if (frase.equals("FIM")) {
                break;
            } else {

                for (int i = 0, j = frase.length() - 1; i < j; i++, j--) {
                    if (frase.charAt(i) != frase.charAt(j)) {
                        ispalind = false;
                    }
                }
                if (ispalind)
                    System.out.println("SIM");
                else
                    System.out.println("NAO");
            }
        }
    }
}

File: bin/palindromo.c
#include <stdio.h>
#include <stdbool.h>

int getlenght(char str[])
{
    int lenght = 0;
    while (str[lenght] != '\0')
        lenght++;
    return lenght;
}

bool isPalind(char str[])
{
    bool palin = true;
    int inicio = 0;
    int fim = getlenght(str) - 1;

    while (inicio < fim)
    {
        if (str[inicio] != str[fim])
        {
            palin = false;
            break;
        }
        else
        {
            inicio++;
            fim--;
        }
    }
    return palin;
}

int main()
{
    while (true)
    {
        char string[50] = "";
        scanf(" %49[^\n]", string); 
        if (string[0] == 'F' && string[1] == 'I' && string[2] == 'M' && string[3] == '\0')
            break;
        if (isPalind(string))
            printf("SIM\n");
        else
            printf("NAO\n");
    }
    return 0;
}
